### PR TITLE
Add initial Golang guidelines

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

Add initial Golang guidelines. They are intended point developers in the right direction when developing Golang apps.

### How to review

Check the new content is appropriate and see if anything is missing.

### Who can review

Anyone but me

### CODE REVIEW

File: guidelines/GOLANG.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+* Use [gometalinter](https://github.com/alecthomas/gometalinter) to identify code issues
+* Prefer using [GoConvey](https://github.com/smartystreets/goconvey) as a test tool
+* Use [govendor](https://github.com/kardianos/govendor) for vendoring dependencies
+
+#### Libraries
+
+* Utilise the [go-ns](https://github.com/ONSdigital/go-ns) library for common functionality. Extract common code from projects and contribute it to go-ns
+    * Use the common logging implementation
+    * Default to the go-ns HTTP server implementation
+        * If you don't ensure your server utilises the common middleware handlers
```
Comment (eldeal) [inline guidelines/GOLANG.md]: Potentially lacking a comma? 'If you don't, ensure...'

File: guidelines/GOLANG.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+#### General
+* Start with the language level recommendations for Golang in [Effective Go](https://golang.org/doc/effective_go.html) 
+and the [code review guidelines](https://github.com/golang/go/wiki/CodeReviewComments)
+* [Listen for OS signals](https://golang.org/pkg/os/signal/) and gracefully shutdown the app when one is received
+
+#### Tooling
+
+* Use [gometalinter](https://github.com/alecthomas/gometalinter) to identify code issues
+* Prefer using [GoConvey](https://github.com/smartystreets/goconvey) as a test tool
+* Use [govendor](https://github.com/kardianos/govendor) for vendoring dependencies
```
Comment (LloydGriffiths) [inline guidelines/GOLANG.md]: Has there been discussions on the use of this tool going forward? Might be worth taking a look at `dep` if not - I believe it's on their roadmap to be integrated into the Go toolchain.

File: guidelines/GOLANG.md (hunk 1 above)
Comment (CarlHembrough) [inline guidelines/GOLANG.md]: I was indeed, thanks 👍

File: guidelines/GOLANG.md (hunk 2 above)
Comment (CarlHembrough) [inline guidelines/GOLANG.md]: There has been no discussions. I was just gathering the current tooling at this stage. It certainly seems like `dep` is worth looking at if its likely to become standard anyway.

Comment (LloydGriffiths) [inline guidelines/GOLANG.md]: Cool, might be something to think about going forward then 👍

File: guidelines/GOLANG.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+#### Libraries
+
+* Utilise the [go-ns](https://github.com/ONSdigital/go-ns) library for common functionality. Extract common code from projects and contribute it to go-ns
+    * Use the common logging implementation
+    * Default to the go-ns HTTP server implementation
+        * If you don't, ensure your server utilises the common middleware handlers
+            * timeout
```
Comment (ian-kent) [inline guidelines/GOLANG.md]: we should probably remove this, the timeout handler provided in go-ns is broken and should be removed

Comment (CarlHembrough) [reply guidelines/GOLANG.md]: Removed the timeout handler from the list. I will remove the actual handler from go-ns.

**Pull request closed without merging**